solana/programs/swap-layer/src/processor/stage_outbound.rs

```diff
@@ -0,0 +1,277 @@
+                SwapLayerError::InvalidTargetChain,
+            );
+
+            true
+        }
+    )]
+    target_peer: RegisteredPeer<'info>,
+
+    /// Staged outbound account, which contains all of the instructions needed to initiate a
+    /// transfer on behalf of the sender.
+    #[account(
+        init,
+        payer = payer,
+        space = StagedOutbound::try_compute_size(&args.redeem_option, &args.encoded_output_token)?,
+        constraint = {
+            // Cannot send to zero address.
+            require!(args.recipient != [0; 32], SwapLayerError::InvalidRecipient);
+
+            true
+        }
+    )]
+    staged_outbound: Account<'info, StagedOutbound>,
+
+    /// Custody token account for the staged outbound transfer. This account will be owned by the
+    /// registered peer.
+    #[account(
+        init,
+        payer = payer,
+        token::mint = src_mint,
+        token::authority = target_peer,
+        seeds = [
+            crate::STAGED_CUSTODY_TOKEN_SEED_PREFIX,
+            staged_outbound.key().as_ref(),
+        ],
+        bump,
+    )]
+    staged_custody_token: Account<'info, token::TokenAccount>,
+
+    #[account(
+        mut,
+        token::mint = common::USDC_MINT,
+    )]
+    usdc_refund_token: Box<Account<'info, token::TokenAccount>>,
+
+    /// Mint can either be USDC or whichever mint is used to swap into USDC.
+    src_mint: Account<'info, token::Mint>,
+
+    token_program: Program<'info, token::Token>,
+    system_program: Program<'info, System>,
+}
+
+/// Arguments for [stage_outbound].
+#[derive(Debug, Clone, AnchorSerialize, AnchorDeserialize)]
+pub struct StageOutboundArgs {
+    pub amount_in: u64,
+
+    /// The Wormhole chain ID of the network to transfer tokens to.
+    pub target_chain: u16,
+
+    /// The recipient of the transfer.
+    pub recipient: [u8; 32],
+
+    pub redeem_option: Option<RedeemOption>,
+
+    pub encoded_output_token: Option<Vec<u8>>,
+}
+
+pub fn stage_outbound(ctx: Context<StageOutbound>, args: StageOutboundArgs) -> Result<()> {
+    // In case we use a program transfer authority, we need to use these for the transfer.
+    let last_transfer_authority_signer_seeds = ctx
+        .bumps
+        .program_transfer_authority
+        .map(|bump| (keccak::hash(&args.try_to_vec().unwrap()).0, bump));
+
+    let StageOutboundArgs {
+        amount_in,
+        target_chain,
+        recipient,
+        redeem_option,
+        encoded_output_token,
+    } = args;
+
+    // Replace None with OutputToken::USDC encoded.
+    let encoded_output_token = encoded_output_token.unwrap_or({
+        let mut buf = Vec::with_capacity(1);
+        OutputToken::Usdc.write(&mut buf).unwrap();
+        buf
+    });
+    let output_token = OutputToken::read(&mut &encoded_output_token[..]).unwrap();
+
+    // We need to determine the relayer fee. This fee will either be paid for right now if
+    // StagedInput::Usdc or will be paid for later if a swap is required to get USDC.
```

Comment:
We should explicitly mention that the relayer fee will be "paid for" by using the swap proceeds.

Reply:
Fixed here: https://github.com/wormholelabs-xyz/swap-layer/pull/30/commits/264000226f6942e5e698d1dcffc5cb1dc2bc033e